### ISSUE TITLE
Fails to install any Ruby version

### ISSUE THREAD

**Issue by stanley90** [asdf-vm/asdf-ruby#267]
I have `asdf` installed by Homebrew. Recently something broke (most likely after some upgrade) and I can't install any Ruby version:

```
$ asdf install ruby 2.7.6
Downloading ruby-build...
Cloning into '/Users/stanley/.asdf/plugins/ruby/ruby-build-source'...
remote: Enumerating objects: 12184, done.
remote: Counting objects: 100% (877/877), done.
remote: Compressing objects: 100% (336/336), done.
remote: Total 12184 (delta 576), reused 732 (delta 485), pack-reused 11307
Receiving objects: 100% (12184/12184), 2.56 MiB | 6.13 MiB/s, done.
Resolving deltas: 100% (8043/8043), done.
Note: switching to 'v20220415'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at d3bbd54 ruby-build 20220415
Warning: unknown option ‘-p’
Error in getOctD(x, offset, len) : invalid octal digit
```

The result is the same regardless of whether I have `ruby-build` installed by Homebrew or not, it always tries to download it (shouldn't it pick up the Homebrew's `ruby-build`?)

**Comment by stanley90** [comment]
Turns out it's not related to Homebrew. The problem is that `ruby_build_path` does not exist.

**Comment by Stratus3D** [comment]
Please provide the output of `asdf info` so we can debug your issue. If you haven't already, try upgrading to the latest version of asdf and asdf-ruby.

**Comment by stanley90** [comment]
I have the latest versions.

```
$ asdf info                                                                                   ✔  1m 39s  2.6.9   10.24.1 
OS:
Darwin fuji 21.4.0 Darwin Kernel Version 21.4.0: Fri Mar 18 00:46:32 PDT 2022; root:xnu-8020.101.4~15/RELEASE_ARM64_T6000 arm64

SHELL:
zsh 5.8 (x86_64-apple-darwin21.0)

ASDF VERSION:
v0.10.0

ASDF ENVIRONMENT VARIABLES:
ASDF_DIR=/opt/homebrew/opt/asdf/libexec

ASDF INSTALLED PLUGINS:
nodejs                       https://github.com/asdf-vm/asdf-nodejs.git master f9957f3
ruby                         https://github.com/asdf-vm/asdf-ruby.git master c841ccd
```

As I said, `ruby_build_path` is empty, it's used in several places, e.g. [here](https://github.com/asdf-vm/asdf-ruby/blob/master/lib/utils.sh#L20)
I tried installing `ruby-build` from both Homebrew and manually from git.

**Comment by stanley90** [comment]
I was hacking it meanwhile and I don't know if I changed something, but now when I `echo $(ruby_build_path)` I get ~/.asdf/plugins/ruby/ruby-build/bin/ruby-build even if I have it installed by Homebrew in `/opt/homebrew/bin/ruby-build`.
Then it gets incorrectly downloaded anyway and fails to build (that would be a problem of `ruby-build`).
So the remaining problem here seems to be that it doesn't find the Homebrew's `ruby-build`.

**Comment by Stratus3D** [comment]
So asdf-ruby downloads and manages its own version of `ruby-build`. You do not need to install it manually yourself using brew. It shouldn't "find" Homebrew's `ruby-build` because it is not looking for it. `~/.asdf/plugins/ruby/ruby-build/bin/ruby-build` is the value I would expect, so I don't believe this has anything to do with your issue.

Did you verify you have installed the dependencies suggested for the build environment? (https://github.com/rbenv/ruby-build/wiki#suggested-build-environment)

**Comment by stanley90** [comment]
Yeah, I get it now. Something in the system must have changed that breaks `ruby-build` install. I might try again when I have the time.

> It shouldn't "find" Homebrew's ruby-build because it is not looking for it

Out of curiosity, why not? My current workaround is to replace `$(ruby_build_path)` with `ruby-build` (which is the Homebrew's `ruby-build`) and then it works.

**Comment by stanley90** [comment]
Yes, that runs the `ruby-build` that I have in path from Homebrew. 
The problem is that the own downloaded version fails to install. I'll close this.
